### PR TITLE
Add more comments and documentation to the codebase

### PR DESCRIPTION
**Pull request by Jasonoro**

### CODE REVIEW

File: src/Synthesizer/Converters/Language.hs [hunk 1]
```diff
@@ -17,17 +17,20 @@ import Synthesizer.Structure           (Channel (..), Frequency, Sample,
                                         SynSound (..))
 
 
+-- | Convert the DSL to a 'SynSound'. This allows the user to add custom sounds to the 'SynSound' if they so wish.
 convertMusicPieceToSynthesizer :: MusicPiece -> SynSound
 convertMusicPieceToSynthesizer musicPiece = SynSound $ map convertInstrumentToChannel instruments
   where
     (MusicPiece instruments) = musicPiece
 
+-- | Converts an instrument block in the DSL to a Synthesizer channel.
```
Comment (jortvangorkum) [inline src/Synthesizer/Converters/Language.hs]: What do you mean by instrument block?

File: src/Synthesizer/Converters/Language.hs [hunk 2]
```diff
@@ -17,17 +17,20 @@ import Synthesizer.Structure           (Channel (..), Frequency, Sample,
...
 convertInstrumentToChannel :: Instrument -> Channel
 convertInstrumentToChannel instrument = Channel $ map (applyEnvelope noteStrike . convertNoteEventsToSoundEvents noteMap baseAmplitude) noteEvents
   where
     (Instrument baseFreq baseAmplitude noteStrike noteEvents) = instrument
     noteMap = generateNotes baseFreq
 
+-- | Convert a single (possibly modified by envelope) note event to a SoundEvent.
 convertNoteEventsToSoundEvents :: Map Note Frequency -> BaseAmplitude -> InstrumentEvent -> SoundEvent
```
Comment (jortvangorkum) [inline src/Synthesizer/Converters/Language.hs]: The comment says something about note event, while the type is InstrumentEvent

File: src/Synthesizer/Modifiers/Envelopes.hs [hunk 3]
```diff
@@ -85,6 +85,7 @@ getSustainSteps sustainLevel eventLength attackLength decayLength samplingRate =
   where
     sustainLength = eventLength - attackLength - decayLength
 
+-- | Calculate the sustain steps based on the release length and the sustain level
```
Comment (jortvangorkum) [inline src/Synthesizer/Modifiers/Envelopes.hs]: I think this calculates the release steps not the sustain steps

**Review by jortvangorkum — CHANGES_REQUESTED** [review]

File: src/Synthesizer/Converters/Language.hs (hunk 1 above)
Comment (Jasonoro) [reply src/Synthesizer/Converters/Language.hs]: The `Instrument freq vol strike [ ... ]` bit of the DSL. I'll elaborate a bit

Comment (Jasonoro) [reply src/Synthesizer/Converters/Language.hs]: Fixed in fcc0c66

File: src/Synthesizer/Converters/Language.hs (hunk 2 above)
Comment (Jasonoro) [reply src/Synthesizer/Converters/Language.hs]: Fixed in fcc0c66

File: src/Synthesizer/Modifiers/Envelopes.hs (hunk 3 above)
Comment (Jasonoro) [reply src/Synthesizer/Modifiers/Envelopes.hs]: Fixed in fcc0c66

**Review by jortvangorkum — APPROVED** [review]

**Pull request merged**